docker/README.md

```diff
@@ -0,0 +1,81 @@
+# What is Docker?
+
+Docker is an open-source platform designed to simplify the development, deployment, and operation of applications by using containerization. Consisting of Docker Engine, a portable, lightweight runtime and packaging tool, and Docker Hub, a cloud service for sharing applications and automating workflows, Docker enables apps to be quickly assembled from components and eliminates the friction between development, QA, and production environments.
+
+---
+
+## Why Use Docker?
+
+Since ROS 1 does not support Ubuntu 22.04 or newer OS versions, and Webots R2024a has dropped support for Ubuntu 20.04 and ROS 1, this setup ensures a unified development environment. It allows ROS 1 and Webots to coexist seamlessly, bypassing OS and Webots version restrictions. Also. with this setup, you can use GUI applications directly from Docker, such as RViz, with GPU acceleration enabled.
+
+---
+
+## Learn more about Docker:
+- [Official Docker Documentation](https://docs.docker.com/)
+- [Docker CLI Reference](https://docs.docker.com/engine/reference/commandline/docker/)
+
+---
+
+## Dependencies
+
+```bash
+sudo apt install git make curl
+curl -sSL https://get.docker.com | sh && sudo usermod -aG docker $USER
+```
+
+## Use Docker with Webots and ROS 1:
+
+This guide explains how to set up and run Docker with Webots and ROS 1 using the `webots_ros` repository.
+
+## Steps
+
+1. Clone the `webots_ros` repository:
+
+    ```bash
+    git clone https://github.com/cyberbotics/webots_ros.git
+    ```
+
+2. Navigate into the `docker` directory:
+
+    ```bash
+    cd ~/webots_ros/docker
+    ```
+
+3.  Run the command (edit the Webots version if needed):
+
+    ```bash
+    WEBOTS_VERSION=2023b make build run exec
+    ```
+
+4.  Build the ROS workspace:
+
+    ```bash
+    catkin_make
+    ```
+5. Source the environment:
+
+    ```bash
+    source devel/setup.bash
+    ```
+6.  Run a simulation with an example launch file:
+
+    ```bash
+    roslaunch webots_ros tiago.launch
+    ```
+7.  Open a new terminal and enter the existing container:
+
+    ```bash
+    make exec
+    ```
+8.  Open RViz:
+
+    ```bash
+    rviz -d ~/ros_ws/src/webots_ros/config/rviz_config.rviz
+    ```
+
```

Comment:
As an example, you can suggest to a user to try teleoperating the robot (it would be a nice interactive demo):
```
make exec
rosrun teleop_twist_keyboard teleop_twist_keyboard.py cmd_vel:=/diff_drive_controller/cmd_vel
```